src/base/Assert.hh

```diff
@@ -60,26 +81,37 @@
 //! \endcond
 
 #if CELERITAS_DEBUG && defined(__CUDA_ARCH__)
-#    define REQUIRE(x) CELER_CUDA_ASSERT_(x)
-#    define CHECK(x) CELER_CUDA_ASSERT_(x)
-#    define ENSURE(x) CELER_CUDA_ASSERT_(x)
+#    define REQUIRE(COND) CELER_CUDA_ASSERT_(COND)
+#    define CHECK(COND) CELER_CUDA_ASSERT_(COND)
+#    define ENSURE(COND) CELER_CUDA_ASSERT_(COND)
 #    define CHECK_UNREACHABLE CELER_CUDA_ASSERT_(false)
 #elif CELERITAS_DEBUG && !defined(__CUDA_ARCH__)
-#    define REQUIRE(x) CELER_ASSERT_(x)
-#    define CHECK(x) CELER_ASSERT_(x)
-#    define ENSURE(x) CELER_ASSERT_(x)
-#    define CHECK_UNREACHABLE CELER_ASSERT_(false)
+#    define REQUIRE(COND) CELER_DEBUG_ASSERT_(COND, precondition)
```

Comment:
Those name are pretty 'generic', shouldn't we use 
```
CELER_REQUIRE
CELER_CHECK
CELER_ENSURE
CELER_CHECK_UNEREACHABLE
```
to avoid potential conflicts?

Reply:
Yeah, I agree that's better -- but at least they're using macro-style case rather than function-style case! 😅 In Shift we ran into a conflict between a macro named `Validate` and a function/enum in Trilinos.

Do you think this change is one that should be started immediately (new find-replace PR with deprecated macros for MRs currently in progress), or one that should wait until Celeritas starts being used as a library for external codes? I would tend toward the latter for the sake of convenience -- cleaning up our "external" interfaces when we can reliably export the library and headers.

Should we tackle this now, or add this to the list of things we can/should do to improve the codebase especially as an external-facing library? 😬

Reply:
(Closing this discussion as it got rolled into https://github.com/celeritas-project/celeritas/pull/95#discussion_r555988955)